common/src/View/Autosaver.cpp

```diff
@@ -135,6 +135,11 @@ namespace TrenchBroom {
                     return false;
                 }
 
+                const auto backupExtension = backupName.extension();
+                if (!StringUtils::isNumber(backupExtension)) {
```

Comment:
Just an idea, turn BackupFileMatcher into a static function e.g. `isBackupFileName`, expose it in the header, and have a couple tests like:
```
EXPECT_TRUE(isBackupFileName("foo.map", "foo.1.map"));
EXPECT_FALSE(isBackupFileName("foo.map", "not-backup.map));
```

Reply:
I have added some separate tests for `BackupFileMatcher`. It cannot easily be turned into a static method because it needs state.